lib/src/transformer/impl_generation.dart

```diff
@@ -104,9 +104,10 @@ class ImplGenerator {
       }
 
       declarations.factory.node.variables.variables.forEach((variable) {
-        if (variable.initializer != null) {
+        if (variable.initializer != null && !(variable.initializer.toString() == '\$$factoryName')) {
           logger.error(
-              'Factory variables are stubs for the generated factories, and should not have initializers.',
+              'Factory variables are stubs for the generated factories, and should not have initializers '
+              'unless initialized with \$<UiFactory> for Dart 2 builder compatibility.',
```

Comment:
I think we can have a more helpful error message here. `UiFactory` is a type so I don't think it should be used as a placeholder here.

Reply:
Yeah, maybe just include the expected string? 
```dart
'unless initialized with \$$factoryName...';
```